udt.go

```diff
@@ -39,26 +39,24 @@ func makeUDT(value reflect.Value, mapper *reflectx.Mapper, unsafe bool) udt {
 
 func (u udt) MarshalUDT(name string, info gocql.TypeInfo) ([]byte, error) {
 	value, ok := u.field[name]
-
-	var data []byte
-	var err error
 	if ok {
-		data, err = gocql.Marshal(info, value.Interface())
-		if err != nil {
-			return nil, err
-		}
+		return gocql.Marshal(info, value.Interface())
 	}
-
-	return data, err
+	if u.unsafe {
+		return nil, nil
+	}
+	return nil, fmt.Errorf("missing name %q in %s", name, u.value.Type())
 }
 
 func (u udt) UnmarshalUDT(name string, info gocql.TypeInfo, data []byte) error {
 	value, ok := u.field[name]
-	if !ok && !u.unsafe {
-		return fmt.Errorf("missing name %q in %s", name, u.value.Type())
+	if ok {
+		return gocql.Unmarshal(info, data, value.Addr().Interface())
 	}
-
-	return gocql.Unmarshal(info, data, value.Addr().Interface())
+	if u.unsafe {
+		return nil
+	}
+	return fmt.Errorf("missing name %q in %s", name, u.value.Type())
```

Comment:
Are we sure this is the semantics of `unsafe` that we want to have?
In Rust driver at least, we consider _too many_ and _too few_ fields in the CQL UDT compared to the go struct as two distinct cases. In Rust driver, _strict_ forbids _too many_ fields in CQL UDT, whereas _default_if_missing_ is used for the case when a struct has a field that is not found in the CQL UDT metadata. WDYT? Should we cover it by a single lever here?

Reply:
@wprzytula, i am totally in support of having all drivers as close as possible to each other. Let's have separate issue on that and address it in other PR

Reply:
OTOH I have no idea how other drivers apart from Rust driver approach this.

Reply:
We can start from these two